libraries/book/src/test/java/com/paritytrading/parity/book/MarketTest.java

```diff
@@ -201,6 +201,7 @@ public void empty() {
                     updateAfterFirstDelete, updateAfterSecondDelete), events.collect());
     }
 
+    @SuppressWarnings("unused")
```

Comment:
What generates these warnings? Would it be possible to configure them in the `pom.xml` for Maven as well?

Reply:
Here's a first-pass https://github.com/JanStureNielsen/parity/tree/code-analysis -- still has some false positives but there might still be some value.

Reply:
Did you implement these changes to see this 'unused' warning, or did you get it, for example, by default on Eclipse (or some other IDE)? I'm asking because some Parity projects used to use [Error Prone](https://errorprone.info/) in the past, but eventually I decided to get rid of it as it was making it more complicated to be able to build the projects both on Java 8 and Java 11.

I noticed from the branch that you have also added quite a few rules to Checkstyle. As you might have guessed, so far the Checkstyle usage has been more about overcoming some limitations when working on the projects on non-IDEs (like my choice of Vim) rather than being very specific about the code formatting. 🙂

Reply:
No; I saw the warnings in Eclipse - one of the default Eclipse plugins does the static analysis and flags the 'unused' and listens to the `@SuppressWarnings` annotations. This branch is an attempt to get similar static analysis working on the command line, well, limiting it to the "Good Things" I see in Eclipse's static analysis. The checkstyle configuration is largely Google's checkstyle with a few tweaks for the "parity" style of things...which also helps me (I split my time between Emacs and Eclipse). It may not be useful to have such an extensive set of analysis tools but I thought I'd play with just to see where I can get...but the tool-chain may not be limber enough to get to Java 11 quickly.